### PR TITLE
Enabling Sound Null Safety

### PR DESCRIPTION
**Pull request by DevNico**
As mentioned in #72 here is my PR. It's probably best to keep this on a separate branch on your end until null safety hits a stable release but I am not sure how to handle that so that's up to you.

You will probably have to change the google/dart container to use the dev tag in order for the checks to work

### CODE REVIEW

File: lib/src/outputs/memory_output.dart [hunk 1]
```diff
@@ -25,8 +24,6 @@ class MemoryOutput extends LogOutput {
 
     buffer.add(event);
 
-    if (secondOutput != null) {
-      secondOutput.output(event);
-    }
+    secondOutput?.output(event);
```
Comment (haarts) [inline lib/src/outputs/memory_output.dart:27]: God. I do love this.

File: lib/src/printers/pretty_printer.dart [hunk 2]
```diff
@@ -45,23 +45,20 @@ class PrettyPrinter extends LogPrinter {
   /// Matches a stacktrace line as generated on Android/iOS devices.
   /// For example:
   /// #1      Logger.log (package:logger/src/logger.dart:115:29)
-  static final _deviceStackTraceRegex =
-      RegExp(r'#[0-9]+[\s]+(.+) \(([^\s]+)\)');
+  static final _deviceStackTraceRegex = RegExp(r'#[0-9]+[\s]+(.+) \(([^\s]+)\)');
```
Comment (haarts) [inline lib/src/printers/pretty_printer.dart:48]: These changes (just formatting?) seem irrelevant for the goal.

File: pubspec.yaml [hunk 3]
```diff
@@ -1,11 +1,11 @@
 name: logger
 description: Small, easy to use and extensible logger which prints beautiful logs.
-version: 0.9.4
+version: 1.0.0-nullsafety.1
```
Comment (haarts) [inline pubspec.yaml:3]: Please leave the package versioning to me.

File: pubspec.yaml [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
 homepage: https://github.com/leisim/logger
 
 environment:
-  sdk: ">=2.2.0 <3.0.0"
+  sdk: ">=2.12.0-0 <3.0.0"
 
 dev_dependencies:
-  test: any
+  test: ^1.16.0-nullsafety.10
   pedantic: ^1.9.0
```
Comment (haarts) [inline pubspec.yaml:11]: `pedantic` has a nullsafety version too.

**Comment by DevNico** [comment]
Would you mind me running `dartfmt .` since the there is a CI check checking if darfmt would change something.

**Comment by haarts** [comment]
Please do

**Comment by haarts** [comment]
Just wondering; did you 'just' run the `dart migrate` tool on the codebase?

**Comment by haarts** [comment]
This PR has a conflict with master which needs to be resolved. Can you also squash the commits into one?

**Comment by DevNico** [comment]
I did not use the migrate tool, will squash the commits tonight.

**Comment by haarts** [comment]
Thank you for your significant contribution!
